janitor/functions.py

```diff
@@ -3815,3 +3816,59 @@ def jitter(
     df[dest_column_name] = result
 
     return df
+
+
+@pf.register_dataframe_method
+def sort_naturally(
+    df: pd.DataFrame, column_name: str, **natsorted_kwargs
+) -> pd.DataFrame:
+    """
+    Sort an DataFrame by a column using "natural" sorting.
```

Comment:
I'm not really familiar with what "natural" sorting provides. Could we add a few examples in the docstring that demonstrate what this would provide (that a normal sort wouldn't)? I think that's probably taken care of in `natsort`'s documentation, but would be helpful here for our users

Reply:
Good comment, @hectormz! I added an example in the docstring.